src/foam/nanos/session/SimpleSessionService.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.session',
+  name: 'SimpleSessionService',
+
+  documentation: `
+    A simple implementation of the SessionService interface. Uses SPID-based
+    access control checks. Allows SPID administrators to create and destroy
+    sessions for users in the SPID they administrate.
+  `,
+
+  implements: ['foam.nanos.session.SessionService'],
+
+  imports: [
+    'auth',
+    'localSessionDAO',
+    'localUserDAO'
+  ],
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.auth.AuthorizationException',
+    'foam.nanos.auth.User',
+    'foam.util.SafetyUtil'
+  ],
+
+  methods: [
+    // Interface methods
+    {
+      name: 'createSession',
+      javaCode: `
+        return createSessionWithTTL(x, userId, 0);
+      `
+    },
+    {
+      name: 'createSessionWithTTL',
+      javaCode: `
+        if ( userId < 1 ) {
+          throw new IllegalArgumentException("User id must be a positive integer.");
+        }
+
+        if ( ! hasSPIDPermission(x, "create", userId) ) {
+          throw new AuthorizationException("You don't have permission to create a session for that user.");
+        }
+
+        Session session = new Session.Builder(x)
+          .setUserId(userId)
+          .build();
+        
+        if ( ttl > 0 ) {
+          session.setTtl(ttl);
+        }
+
+        session = (Session) ((DAO) getLocalSessionDAO()).put(session);
+
+        // TODO: Change to access token property when we support that.
+        return session.getId();
+      `
+    },
+    {
+      name: 'destroySession',
+      javaCode: `
+        DAO localSessionDAO = (DAO) getLocalSessionDAO();
+
+        // TODO: Change to query by access token property instead of id when we
+        // support that.
+        Session session = (Session) localSessionDAO.find(accessToken);
+
+        if ( session == null ) {
+          throw new RuntimeException("Session not found.");
+        }
+
+        if (
```

Comment:
Is this check actually preventing anything? If I know the sessionID, I could just use it to logout the user anyway (or do anything else on behalf of the user).

Reply:
Hmm, good point. I think you're right.

What if this method took a user id instead and deleted all sessions for that user? Then the check would make sense again since guessing or obtaining another user's id is relatively easy since they're sequence numbers and we don't make any effort to hide user ids, whereas guessing someone's session id (a GUID) is (intended to be) impossible.

Reply:
I think we would only want to let you delete sessions you created. If the user logs into the system themself do we want to let you kick them off? If you want to disable them, and you have the access, then you could disable the User and the system could then decide if it wants to remove their active sessions or the Auth system would just stop let them doing things anyway.

Reply:
Fair enough. I'll just remove this method then.

Reply:
Resolved by e8a9568.